### PR TITLE
feat(litestream): initialise

### PR DESCRIPTION
**Pull request by hlolli**

### CODE REVIEW

File: litestream/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM litestream/litestream:0.3.13
```
Comment (dtfiedler) [inline litestream/Dockerfile:1]: this image does not have `ytt` or curl

you can do something like this to load `ytt` into the container bin and use it to map env values at run time

https://github.com/ar-io/arns-service/blob/litestream/litestream/Dockerfile

File: litestream/litestream.template.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - path: /app/data/sqlite/core.db
+    replicas:
+      - type: s3
+        endpoint: #@ data.values.AR_IO_SQLITE_BACKUP_S3_BUCKET_URL
+        name: ar_io_node_core.backup.db
+        bucket: #@ data.values.AR_IO_SQLITE_BACKUP_S3_BUCKET_NAME
+        region: #@ data.values.AR_IO_SQLITE_BACKUP_S3_BUCKET_REGION
+        forcePathStyle: true
+        sync-interval: 1s
+        access-key-id: #@ data.values.AR_IO_SQLITE_BACKUP_S3_BUCKET_ACCESS_KEY
```
Comment (djwhitt) [inline litestream/litestream.template.yaml:11]: I'd ideally like the ability to use the EC2 instance role here if no explicit keys are used. That probably the default if the values aren't set explicitly. Putting a conditional around this part of the template based on whether `data.values.AR_IO_SQLITE_BACKUP_S3_BUCKET_ACCESS_KEY` is set should accomplish that.

File: docker-compose.yaml [hunk 3]
```diff
@@ -1,5 +1,72 @@
----
-version: '3.0'
+version: '3.8'
```
Comment (djwhitt) [inline docker-compose.yaml:1]: As long as these changes still work, let's leave this at `3.0`.

File: docker-compose.yaml [hunk 4]
```diff
@@ -1,5 +1,72 @@
...
+
+volumes:
```
Comment (djwhitt) [inline docker-compose.yaml:3]: I like this `volumes` section. Did you do any testing with it to confirm bind mounts still work as expected?

File: litestream/litestream.template.yaml (hunk 2 above)
Comment (djwhitt) [reply litestream/litestream.template.yaml:11]: I also wonder if these can be set in such a way that they'd apply to multiple replicas when we add them.

**Review by djwhitt — COMMENTED** [review]
This is looking pretty good. Have you gotten it to successfully write to S3?

File: litestream/litestream.template.yaml (hunk 2 above)
Comment (dtfiedler) [reply litestream/litestream.template.yaml:11]: litestream will use the instance creds by default - you do not need to set these

Comment (djwhitt) [reply litestream/litestream.template.yaml:11]: Nice. I think we do still want support for setting them though. There are probably some users who will run on their own hardware but still backup to S3.

File: docker-compose.yaml (hunk 3 above)
Comment (hlolli) [reply docker-compose.yaml:1]: sure, it seems furthermore that version is itself obsolete

```
$ docker compose --profile litestream up
WARN[0000] /home/ubuntu/ar-io-node/docker-compose.yaml: `version` is obsolete
```

File: litestream/Dockerfile (hunk 1 above)
Comment (hlolli) [reply litestream/Dockerfile:1]: fixed

File: docker-compose.yaml (hunk 4 above)
Comment (hlolli) [reply docker-compose.yaml:3]: it's now been tested, and it works after making sure all mountpoints exist

File: litestream/litestream.template.yaml (hunk 2 above)
Comment (djwhitt) [reply litestream/litestream.template.yaml:12]: The new conditional looks good. I think we should still probably do this though:
> I also wonder if these can be set in such a way that they'd apply to multiple replicas when we add them.
In other words, we should make those global settings.

Comment (hlolli) [reply litestream/litestream.template.yaml:12]: good idea, will implement

Comment (hlolli) [reply litestream/litestream.template.yaml]: done and I verified that it's still working with these keys in top-level scope

**Review by djwhitt — COMMENTED** [review]
This looks good. I'm going to go ahead and contradict the original ticket a bit and say let's go ahead and add all the DBs (`core.db`, `bundles.db`, `data.db`, `moderation.db`) to the config since it seems easy. I was originally thinking we might need a separate service for each DB.

File: docker-compose.yaml [hunk 5]
```diff
@@ -1,6 +1,74 @@
...
+    driver_opts:
+      type: none
+      o: bind
+      device: ${WARP_CACHE_PATH:-./data/warp/cache}
+  wallets-data:
+    driver: local
+    driver_opts:
+      type: none
+      o: bind
+      device: ${WALLETS_PATH:-./wallets}
```
[the remark this answers is not in the document]
Comment (hlolli) [reply docker-compose.yaml:70]: fixed

**Pull request closed without merging**